src/components/MapTiff/MapTiff.tsx

```diff
@@ -149,10 +148,25 @@ const MapTiff = ({
           });
         }
       }
+    },
+    [map, mapsData, setLoading],
+  );
+
+  const cacheMapData = useCallback(async () => {
+    mapsData.forEach(async (data) => {
+      const id = data.fields.id;
+      const imageData = data.fields.imageData;
+      Object.keys(imageData).forEach(async (element) => {
+        await loadSource(id, element);
+      });
+    });
+  }, [mapsData, loadSource]);
```

Comment:
```suggestion
  const cacheMapData = useCallback( () => {
    mapsData.forEach(async (data) => {
      const id = data.fields.id;
      const imageData = data.fields.imageData;
      if (!isReduced) {
          Object.keys(imageData).forEach(async (year) => {
             loadSource(id, year);
          });
      } else {
          const datas = Object.keys(imageData);
          loadSource(id, datas[datas.length - 1]);
      }
    });
  }, [mapsData, loadSource, isReduced]);
```
i think we could make this function asynchronous. it would allow the lighter packages to be loaded first, right?! not pretty sure how this is supposed to make and order the requests. at least, while testing with asynchronous version of the code, the lighter ones always got requested or at least their request got resolved first. i think this could improve the loading process. this would be even more benefficial to our application because we start by default in a lighter visualization ("cisternas"), so, making this changes, it would load it first.
Also, we could limit the data we request when we're dealing with the home page map. In this page, we should only request the last year for the visualizations which have a yearly display option.